src/gui/dialogs/addon/addon_auth.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+   Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+#define GETTEXT_DOMAIN "wesnoth-lib"
+
+#include "gui/dialogs/addon/addon_auth.hpp"
+
+#include "gui/auxiliary/find_widget.hpp"
+#include "gui/widgets/password_box.hpp"
+#include "gui/widgets/window.hpp"
+
+namespace gui2::dialogs
+{
+
+REGISTER_DIALOG(addon_auth)
+
+addon_auth::addon_auth()
+{
+	password_ = register_text("password", true);
```

Comment:
What do you do with this?

Reply:
It's the password field used to enter the password to send when uploading an add-on, instead of taking it from the _server.pbl.

Reply:
If you don't use the `password_` field you don't need to save a pointer in this class.

Reply:
Alright, will fix.